### PR TITLE
Package LabVIEW Memory Management Tools in a separate package

### PR DESCRIPTION
**Pull request by Karl-G1**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-custom-device-development-tools/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Package LabVIEW Memory Management Tools in a separate package

### Why should this Pull Request be merged?

Enable easier distribution for LVMM component without requiring VeriStand be installed.

### What testing has been done?

Verified PR build

### CODE REVIEW

File: control_lvmm [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Package: labview-memory-management-tools-{labview_version}-support
```
Comment (buckd) [inline control_lvmm:1]: The NIPM convention is to have <product>-<version>-support, so
`labview-memory-management-tools-labview-{labview_version}-support`.

File: build.toml [hunk 2]
```diff
@@ -82,7 +82,14 @@ project = '{network_comm}'
...
 type = 'nipkg'
+control_file = 'control'
 payload_dir = 'Built'
 install_destination = 'ni-paths-LV{labview_version}DIR'
+
+[[package]]
+type = 'nipkg'
+control_file = 'control_lvmm'
+payload_dir = 'lv-memory-manager\Built'
+install_destination = 'ni-paths-LV{labview_version}DIR'
```
Comment (buckd) [inline build.toml:95]: Add a newline to the end of `build.toml` per convention.

File: control_lvmm (hunk 1 above)
Comment (buckd) [reply control_lvmm:1]: Also, the package name should start with the company.

`ni-labview-memory-management-tools-labview-{labview_version}-support`

**Comment by Karl-G1** [comment]
@buckd I corrected the couple fixes you suggested and published the PR for review. 
My biggest open question is if we should 1) conflict with the full dev tools package or 2) exclude LVMM from the full dev tools package.

**Comment by rtzoeller** [comment]
> @buckd I corrected the couple fixes you suggested and published the PR for review.
> My biggest open question is if we should 1) conflict with the full dev tools package or 2) exclude LVMM from the full dev tools package.

If we make them conflict, we should also make the dev tools package "provide" the LVMM package name, so it can replace it in a dependency chain. But I would prefer to exclude LVMM from the main package, and not have duplicate binaries between them.

**Comment by Karl-G1** [comment]
I've removed LVMM from the ni-custom-device-development-tools package by removing the copy step from the post-build action VI. I've also added documentation about the package split to the readmes.

File: README.md [hunk 3]
```diff
@@ -35,20 +35,23 @@ The following dependencies are used by some or all of the included components of
...
+|Error API|error-api|https://github.com/NIVeriStandAdd-Ons/VeriStand-Error-API|[e9d3eb2](https://github.com/NIVeriStandAdd-Ons/VeriStand-Error-API/commit/e9d3eb2148d2cab9452d25e1ea2b2a120602deb5)|2020-07-02|Common|
+|LabVIEW Memory Manager|lv-memory-manager|https://github.com/ni/labview-memory-management-tools|[4a542e2](https://github.com/ni/labview-memory-management-tools/commit/4a542e226cecc1243db6d32982f74b91b5c07874)|2020-06-22|ni-labview-memory-management-tools
 
 ### Releases
 This repository will regularly release builds of the dependencies using NI Packages per LabVIEW version. Custom device developers and open-source contributors should look for updated dependencies before submitting code to custom device repositories on github.com/ni.
 
+### Packages
+Most of the components are included in a common NI Package (`ni-custom-device-development-tools`). Some components are packaged individually, such as the LabVIEW Memory Manager tools. See the table above for which package contains each component.
```
Comment (rtzoeller) [inline README.md:53]: I feel like this should go above the table. That would also solve the back-referencing, i.e. you could write "See the table below...".

File: build.toml [hunk 4]
```diff
@@ -82,7 +82,16 @@ project = '{network_comm}'
 target = 'My Computer'
 build_spec = 'Move Source'
 
-[package]
+[[package]]
 type = 'nipkg'
+control_file = 'control'
 payload_dir = 'Built'
 install_destination = 'ni-paths-LV{labview_version}DIR'
+package_output_dir = 'Built'
```
Comment (rtzoeller) [inline build.toml:90]: You don't need this line.

Comment (Karl-G1) [reply build.toml]: I had just included it for readability. Removed now.

File: README.md (hunk 3 above)
Comment (Karl-G1) [reply README.md]: Updated to move comment above table.

**Review by rtzoeller — APPROVED** [review]

**Review by buckd — APPROVED** [review]

**Comment by Karl-G1** [comment]
Before merging, I noticed that the files for LVMM were not being archived along with the artifacts from the rest of the builds. Talking with @rtzoeller, the easiest solution was to store other builds in Built\Common and LVMM in Built\LMM. This way, we will build packages out of those locations and persist all files. The latest build will test these changes.

**Pull request merged**